packages/rum/src/domain/record/startFullSnapshots.ts

```diff
@@ -64,18 +65,33 @@ export function startFullSnapshots(
     }
     return records
   }
-
   fullSnapshotCallback(takeFullSnapshot())
-
+  let cancelIdleCallback: (() => void) | undefined
   const { unsubscribe } = lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (view) => {
     flushMutations()
-    fullSnapshotCallback(
-      takeFullSnapshot(view.startClocks.timeStamp, {
-        shadowRootsController,
-        status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
-        elementsScrollPositions,
+    if (isExperimentalFeatureEnabled(ExperimentalFeature.ASYNC_FULL_SNAPSHOT)) {
+      if (cancelIdleCallback) {
+        cancelIdleCallback()
+      }
+
+      cancelIdleCallback = requestIdleCallback(() => {
+        fullSnapshotCallback(
+          takeFullSnapshot(view.startClocks.timeStamp, {
+            shadowRootsController,
+            status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
+            elementsScrollPositions,
+          })
+        )
       })
-    )
+    } else {
+      fullSnapshotCallback(
+        takeFullSnapshot(view.startClocks.timeStamp, {
+          shadowRootsController,
+          status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
+          elementsScrollPositions,
+        })
+      )
+    }
```

Comment:
**🔨 warning:** ‏`flushMutations()` should be done directly before `takeFullSnapshot()`

**💬 suggestion:** ‏factorize duplicated code

```ts
    function takeSubsequentFullSnapshot() {
      flushMutations()
      fullSnapshotCallback(
        takeFullSnapshot(view.startClocks.timeStamp, {
          shadowRootsController,
          status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
          elementsScrollPositions,
        })
      )
    }

    if (isExperimentalFeatureEnabled(ExperimentalFeature.ASYNC_FULL_SNAPSHOT)) {
      if (cancelIdleCallback) {
        cancelIdleCallback()
      }

      cancelIdleCallback = requestIdleCallback(takeSubsequentFullSnapshot)
    } else {
      takeSubsequentFullSnapshot()
    }
```